packages/frontend/src/features/metricsCatalog/hooks/useRunMetricExplorerQuery.ts

```diff
@@ -77,18 +77,23 @@ export const useRunMetricExplorerQuery = ({
             projectUuid,
             exploreName,
             metricName,
-            comparison?.type ?? 'none',
-            dateRange?.[0] ?? 'none',
-            dateRange?.[1] ?? 'none',
+            dateRange![0],
+            dateRange![1],
+            comparison!.type,
         ],
         queryFn: () =>
             postRunMetricExplorerQuery({
                 projectUuid: projectUuid!,
                 exploreName: exploreName!,
                 metricName: metricName!,
                 comparison: comparison!,
-                dateRange,
+                dateRange: dateRange!,
             }),
-        enabled: !!projectUuid && !!exploreName && !!metricName && !!comparison,
+        enabled:
+            !!projectUuid &&
+            !!exploreName &&
+            !!metricName &&
+            !!comparison &&
+            !!dateRange,
```

Comment:
```suggestion
            !!dateRange[0],
            !!dateRange[1],
```

<img width="921" alt="image" src="https://github.com/user-attachments/assets/dbfea751-12d7-441d-bd85-377d69774a10">